io/zenoh-transport/src/unicast/universal/rx.rs

```diff
@@ -97,19 +97,19 @@ impl TransportUnicastUniversal {
             Reliability::BestEffort => zlock!(c.best_effort),
         };
 
-        self.verify_sn(sn, &mut guard)?;
-
-        let callback = zread!(self.callback).clone();
-        if let Some(callback) = callback.as_ref() {
-            for msg in payload.drain(..) {
-                self.trigger_callback(callback.as_ref(), msg)?;
+        if self.verify_sn(sn, &mut guard)? {
```

Comment:
In general I find a high level of nesting harder to read than multiple early returns.
E.g.:
```rust
if cond1 {
  if cond2 {
    // Do stuff
  } else {
    println!("Cond 2 not met"):
  }
} else {
  println!("Cond 1 not met");
}
```
*vs*
```rust
if !cond1! {
  println!("Cond 1 not met);
  return Err(());
}
if !cond2! {
  println!("Cond 2 not met);
  return Err(());
}
// Do stuff
```